SQL-1285: Remove ANSI endpoint

nbagnard opened the pull request:
Where to find the list of W functions => https://learn.microsoft.com/en-us/sql/odbc/reference/develop-app/unicode-function-arguments

terakilobyte reviewed (approved):
Changes LGTM. Looks like clippy and rustfmt are being tyrants.

jchemburkar commented on odbc/src/api/diag.rs:
```diff
@@ -40,34 +25,6 @@ pub unsafe fn set_sql_statew(sql_state: &str, output_ptr: *mut WideChar) {
-/// # Safety
-/// This writes to multiple raw C-pointers
-///
-pub unsafe fn get_diag_rec(
```
This looks good! I think we can simplify the `get_diag_field` function now as well to remove the `is_wstring` flag

jchemburkar commented on odbc/src/api/functions.rs:
```diff
@@ -663,77 +602,6 @@ pub unsafe extern "C" fn SQLColumnPrivilegesW(
-///
-#[named]
-#[no_mangle]
-pub unsafe extern "C" fn SQLColumns(
```
For these, and the other functions that _were_ implemented and now removed, should we add them to the list of `Unsupported Operators` in the README?

nbagnard replied:
Good call!
I looked at the README and I don't really see a need to go into this level of details. I'll add a line stating that the driver does not support ANSI endpoints, that should be enough, no need to list all the ANSI functions.
But, I still need to also clean-up the list of unsupported functions.

jchemburkar reviewed (approved):
LGTM! Just had a couple small thoughts

nbagnard commented:
@jchemburkar, @terakilobyte : I updated the README, do you mind giving it one last look? I re-requested your review to make sure that it makes sense to you. I think it's simpler to state that we don't support ANSI functions and then list the ones that we don't support at all. But let me know if that is not working for you. I also cleaned the README and removed the 32-bit steps because we don't support 32-bit.
I left the registry update scripts because if people are coming to this page, they won't have access to the installer.

terakilobyte commented on odbc/src/api/diag.rs:
```diff
@@ -150,41 +106,25 @@ pub unsafe fn get_diag_field(
-                        ),
-                    },
+                    DiagType::SQL_DIAG_SQLSTATE => i16_len::set_output_wstring_as_bytes(
+                        error.get_sql_state(),
+                        diag_info_ptr,
+                        buffer_length as usize,
+                        string_length_ptr,
+                    ),
```
Not related to this PR at all, but I'd really like refactoring all the string writing to cstr. I could create a ticket. WDYT?

nbagnard replied:
> Not related to this PR at all, but I'd really like refactoring all the string writing to cstr. I could create a ticket. WDYT?

@terakilobyte Please do :)

terakilobyte reviewed (approved):
LGTM. Left one small "correction", no need to re-tag if you accept it!

jchemburkar reviewed (approved):
The updates to the README look good to me!

nbagnard commented:
evergreen merge
